photosynthesis_metrics/tv.py

```diff
@@ -0,0 +1,153 @@
+r"""Implemetation of Total Variation metric, based on article
+ remi.flamary.com/demos/proxtv.html and www.wikiwand.com/en/Total_variation_denoising
+"""
+
+from typing import Union, Tuple, Optional
+import functools
+import torch
+from torch.nn.modules.loss import _Loss
+
+from .utils import _adjust_dimensions, _validate_input
+
+
+def _compute_tv(x: torch.Tensor, size_average: bool = True, reduction_type: str = 'l2') -> torch.Tensor:
+    r"""Compute Total Variation metric
+    See www.wikiwand.com/en/Total_variation_denoising for additional info about reduction types
+
+    Args:
+        x: Tensor of shape :math:`(N, C, H, W)` holding an input image.
+        size_average: If size_average=True, total variation of all images will be averaged as a scalar.
+        reduction_type: {'l1', 'l2', 'l2_squared'}, defines which type of norm to implement, isotropic  or anisotropic.
+
+    Returns:
+        tv : Total variation of a given tensor
+    """
+
+    assert isinstance(x, torch.Tensor), f'Image stack must be of type torch.Tensor, got {type(x)} instead.'
```

Comment:
Why did you decide to do this assertion here? 
If this function is called from the interface (`def total_variation`), then this and other requirements have to be checked in the interface (that is its purpose).
If this function may be called externally, it should not be protected (single underscore before function name).

Reply:
This is protected function, assert here is not necessary. Removed it